journalpump/rsyslog.py

```diff
@@ -96,7 +96,8 @@ def __init__(
         if protocol is None:
             protocol = "PLAINTEXT"
         if cacerts is not None or protocol == "SSL":
-            self.ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+            self.ssl_context = ssl.create_default_context()
+            self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
             self.ssl_context.verify_mode = ssl.CERT_REQUIRED
```

Comment:
Looking at the source code this line is redundant (but no harm leaving it for clarity) but one thing that does change here is that `check_hostname` is set to `True`, which is sensible setting in general but we did not have that  before so not sure if something will break.

Reply:
good catch! Maybe it is better/safer then to go with 

```python
self.ssl_context = ssl.SSLContext()
self.ssl_context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1
# and (probably not necessary)
# self.ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
```